lightning-graph-sync/src/processing.rs

```
@@ -0,0 +1,591 @@
+	} else {
+		0
+	};
+	let default_htlc_maximum_msat = if tentative_default_htlc_maximum_msat == u64::MAX {
+		OptionalField::Absent
+	} else {
+		OptionalField::Present(tentative_default_htlc_maximum_msat)
+	};
+
+	for _current_update_index in 0..update_count {
+		// println!("update index: {}", _current_update_index);
+
+		let short_channel_id = if let Some(previous_scid) = previous_scid {
+			let scid_delta: BigSize = Readable::read(read_cursor)?;
+			previous_scid + scid_delta.0
+		} else {
+			Readable::read(read_cursor)?
+		};
+		previous_scid = Some(short_channel_id);
+
+		// let short_channel_id: u64 = Readable::read(read_cursor)?;
+		let channel_flags: u8 = Readable::read(read_cursor)?;
+
+		// flags are always sent in full, and hence always need updating
+		let standard_channel_flags = channel_flags & 0b_0000_0011;
+
+		let mut synthetic_update = if channel_flags & 0b_1000_0000 == 0 {
+			// full update
+			// println!("channel flags: {:b}, scid: {}, full update", channel_flags, short_channel_id);
+			UnsignedChannelUpdate {
+				chain_hash,
+				short_channel_id,
+				timestamp: emulated_last_seen_timestamp,
+				flags: standard_channel_flags,
+				cltv_expiry_delta: default_cltv_expiry_delta,
+				htlc_minimum_msat: default_htlc_minimum_msat,
+				htlc_maximum_msat: default_htlc_maximum_msat.clone(),
+				fee_base_msat: default_fee_base_msat,
+				fee_proportional_millionths: default_fee_proportional_millionths,
+				excess_data: vec![],
+			}
+		} else {
+			// incremental update
+			// println!("channel flags: {:b}, incremental update", channel_flags);
+
+			let read_only_network_graph = network_graph.read_only();
+			let channel = read_only_network_graph.channels().get(&short_channel_id);
+			let channel = if let Some(channel) = channel {
+				channel
+			} else {
+				return Err(LightningError {
+					err: "Couldn't find channel for update".to_owned(),
+					action: ErrorAction::IgnoreError,
+				}
+					.into());
+			};
```

FWIW, wherever you have an `Option` and need to return an error for `None` you can simplify with `ok_or_else` and `?`, e.g.:

```rust
			let channel = read_only_network_graph
				.channels()
				.get(&short_channel_id)
				.ok_or_else(|| LightningError {
					err: "Couldn't find channel for update".to_owned(),
					action: ErrorAction::IgnoreError,
				})?;

```